lib/ansible/modules/windows/win_domain_user.ps1

```diff
@@ -0,0 +1,278 @@
+            Set-ADAccountPassword -Identity $username -Reset:$true -Confirm:$false -NewPassword $secure_password
+            $user_obj = Get-ADUser -Identity $username -Properties *
+            $result.password_updated = $true
+            $result.changed = $true
+        }
+
+        # Configure password policies
+        If (($password_never_expires -ne $null) -and ($password_never_expires -ne $user_obj.PasswordNeverExpires)) {
+            Set-ADUser -Identity $username -PasswordNeverExpires $password_never_expires
+            $user_obj = Get-ADUser -Identity $username -Properties *
+            $result.changed = $true
+        }
+        If (($password_expired -ne $null) -and ($password_expired -ne $user_obj.PasswordExpired)) {
+            Set-ADUser -Identity $username -ChangePasswordAtLogon $password_expired
+            $user_obj = Get-ADUser -Identity $username -Properties *
+            $result.changed = $true
+        }
+        If (($user_cannot_change_password -ne $null) -and ($user_cannot_change_password -ne $user_obj.CannotChangePassword)) {
+            Set-ADUser -Identity $username -CannotChangePassword $user_cannot_change_password
+            $user_obj = Get-ADUser -Identity $username -Properties *
+            $result.changed = $true
+        }
+
+        # Assign other account settings
+        If (($upn -ne $null) -and ($upn -ne $user_obj.UserPrincipalName)) {
+            Set-ADUser -Identity $username -UserPrincipalName $upn
+            $user_obj = Get-ADUser -Identity $username -Properties *
+            $result.changed = $true
+        }
+        If (($description -ne $null) -and ($description -ne $user_obj.Description)) {
+            Set-ADUser -Identity $username -description $description
+            $user_obj = Get-ADUser -Identity $username -Properties *
+            $result.changed = $true
+        }
+        If ($enabled -ne $user_obj.Enabled) {
+            Set-ADUser -Identity $username -Enabled $enabled
+            $user_obj = Get-ADUser -Identity $username -Properties *
+            $result.changed = $true
+        }
+        If ((-not $account_locked) -and ($user_obj.LockedOut -eq $true)) {
+            Unlock-ADAccount -Identity $username
+            $user_obj = Get-ADUser -Identity $username -Properties *
+            $result.changed = $true
+        }
+
+        # Set user information
+        Foreach ($key in $user_info.Keys) {
+            If ($user_info[$key] -eq $null) {
+                continue
+            }
+            $value = $user_info[$key]
+            If ($value -ne $user_obj.$key) {
+                $expression = "Set-ADUser -Identity $username -$key '$value'"
+                Invoke-Expression $expression
+                $result.changed = $true
+                $user_obj = Get-ADUser -Identity $username -Properties *
+            }
+        }
+
+        # Configure group assignment
+        If ($groups -ne $null) {
+            If ($groups -is [System.String]) {
```

Comment:
If using the `list` type it will automatically convert it to a list and split it by `,` if it is a string.